code/game/objects/items/candle.dm

```diff
@@ -69,6 +70,32 @@
 		processing_objects.Add(src)
 		update_icon()
 
+/obj/item/candle/proc/flicker(var/amount = rand(5, 15))
+	if(flickering)
+		return
+	flickering = 1
+	spawn(0)
```

Comment:
is this spawn necessary...? the function ends

Reply:
I assume so, that's how it was implemented on the lighting.dm file.

Reply:
it doesn't appear to do anything at all here

Reply:
tested it, the proc doesn't compile if the spawn ain't there

Reply:
What's the error message?

Reply:
`code\game\objects\items\candle.dm:41:error: : invalid expression` on the line that only has `if(lit)`, which does work normally if the spawn(0) is there

Reply:
wait nevermind, it works after fixing some weird indentation issues on my side
removing